### PR TITLE
feat(action-bar, action-pad): Add setFocus method for the expand toggle #974

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #974

## Summary

feat(action-bar, action-pad): Add setFocus method for the expand toggle #974

### CODE REVIEW

**Comment by driskull** [comment]
@jcfranco can you review?

File: src/components/calcite-action-bar/calcite-action-bar.e2e.ts [hunk 1]
```diff
@@ -151,4 +151,23 @@ describe("calcite-action-bar", () => {
...
+  it("should focus on toggle button", async () => {
+    const page = await newE2EPage({
+      html: `<calcite-action-bar>
+      <calcite-action-group>
+        <calcite-action text="Add" icon="plus"></calcite-action>
+      </calcite-action-group>
+    </calcite-action-bar>`
+    });
+
+    const tagName = await page.evaluate(async () => {
```
Comment (jcfranco) [inline src/components/calcite-action-bar/calcite-action-bar.e2e.ts:164]: Did you try the `focusable` test helper? If not, I think I can add some options to make it work for these tests.

File: src/components/calcite-action-bar/calcite-action-bar.tsx [hunk 2]
```diff
@@ -124,6 +127,22 @@ export class CalciteActionBar {
...
+  // --------------------------------------------------------------------------
+  //
+  //  Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(focusId?: "expand-toggle"): Promise<void> {
+    if (focusId === "expand-toggle") {
+      await this.expandToggleEl?.setFocus();
```
Comment (jcfranco) [inline src/components/calcite-action-bar/calcite-action-bar.tsx:139]: Can you use the `focusElement` helper for the `setFocus` methods?

File: src/components/functional/CalciteExpandToggle.tsx [hunk 3]
```diff
@@ -31,13 +32,23 @@ export function toggleChildActionText({
...
-): HTMLCalciteActionElement => {
-  if (tooltip) {
-    tooltip.referenceElement = !expanded && referenceElement;
+const setTooltipReference = ({
+  tooltipExpand,
+  referenceElement,
+  expanded,
+  ref
+}: {
+  tooltipExpand: HTMLCalciteTooltipElement;
```
Comment (jcfranco) [inline src/components/functional/CalciteExpandToggle.tsx:41]: Nitpick: What's the `Expand` part of `tooltipExpand`? `tooltip` seems simpler, but I could be missing something.

**Review by jcfranco — APPROVED** [review]
Other than a few small comments, this LGTM!

File: src/components/calcite-action-bar/calcite-action-bar.e2e.ts (hunk 1 above)
Comment (driskull) [reply src/components/calcite-action-bar/calcite-action-bar.e2e.ts:164]: yeah it wasn't working correctly

File: src/components/functional/CalciteExpandToggle.tsx (hunk 3 above)
Comment (driskull) [reply src/components/functional/CalciteExpandToggle.tsx]: renamed these confusing props :)

File: src/components/calcite-action-bar/calcite-action-bar.e2e.ts (hunk 1 above)
Comment (driskull) [reply src/components/calcite-action-bar/calcite-action-bar.e2e.ts:164]: The focusable test helper assumes that the focused tag should be the same component, but in this case I want to make sure the action is focused.

The focusable helper would need to be enhanced to take a setFocus argument and an additional tag selector.

Comment (jcfranco) [reply src/components/calcite-action-bar/calcite-action-bar.e2e.ts:164]: I'll submit a PR for this. 💪🏼

Comment (driskull) [reply src/components/calcite-action-bar/calcite-action-bar.e2e.ts:164]: Thank you kind sir! 💪

**Pull request merged**